File: content/code/python-jinja2-vs-go-html-template/extends/extends.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

import jinja2
import os


JINJA_ENVIRONMENT = jinja2.Environment(
    loader=jinja2.FileSystemLoader(os.path.dirname(__file__)),
    extensions=['jinja2.ext.autoescape'],
    autoescape=True)


if __name__ == '__main__':
  template_values = {
    'name': 'world',
  }
  template = JINJA_ENVIRONMENT.get_template('index.html')
  print(template.render(template_values))

File: content/code/python-jinja2-vs-go-html-template/extends/extends.go
package main

import (
	"html/template"
	"os"
)

func main() {
	// cannot put base-go.html before index-go.html
	// the following will give empty output
	//t, _ := template.ParseFiles("base-go.html", "index-go.html")
	t, _ := template.ParseFiles("index-go.html", "base-go.html")
	name := "world"
	t.Execute(os.Stdout, name)
}
